src/payment/payment.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { StripeService } from '../stripe/stripe.service';
+import { Status } from '@prisma/client';
+import { GlobalException } from '../exceptions/global.exception';
+import ErrorsTypes from '../errors/errors.enum';
+
+@Injectable()
+export class PaymentService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly stripeService: StripeService
+  ) {}
+
+  async processPayment(userId: string, totalAmount: number, paymentOption: string) {
+    try {
+      const bookingDetails = await this.prismaService.booking.findMany({
+        where: { userId, status: 'PENDING' },
+      });
+
+      const paymentIntent = await this.stripeService.createPaymentIntent(
+        Math.round(totalAmount * 100)
+      );
+
+      await this.prismaService.payment.create({
+        data: {
+          type: paymentOption,
+          transactionId: paymentIntent.id,
+          totalAmount,
+          status: Status.PENDING,
```

Comment:
when status updates? what about booking status?

Reply:
I think I forgot this.